public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -169,6 +170,42 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
     throw new Error('Snapshot not found');
   }
 
+  public restoreDashboardStateFromLocalStorage(dashboard: DashboardScene) {
```

Comment:
wonder if this should be it's own function & file (not sure,  the state manager feels like an ok home and is not very large, but might get lost among the the other tests for dashboards state manager